force-app/main/default/classes/SOM_TestDataFactory.cls

```diff
@@ -540,7 +541,7 @@ public class SOM_TestDataFactory {
 
         createOrderActionInput.order = new SOM_TestDataInput.OrderInput();
         createOrderActionInput.order.status = SOM_OrderStatus.DRAFT.enumValue;
-        createOrderActionInput.order.effectiveDate = Date.today();
+        createOrderActionInput.order.effectiveDate = Date.today().addDays(-3);
```

Comment:
this method will be used by many classes and tests, you should not put this as a fixed value. You can create an overload version of the method createOrderAction passing a Date parameter that will be checked if it is different than null, you assign it in the createOrderActionInput.order.effectiveDate, else you continue to use the Date.today(), so in your test method, you calculate the date you want before passing it to the createOrderAction.